collections/list.go

```diff
@@ -161,3 +195,14 @@ func (l List[T]) String() string {
 func (_ List[T]) Type() CollectionType {
 	return TypeList
 }
+
+// Temporary value store that stores results of Map method.
+type ListTransformation struct {
+	values []interface{}
+}
+
+// Type of callback function that needs to be passed to Map method.
+type listMapFunction func(T any, index int) interface{}
```

Comment:
```suggestion
type listMapFunction[T CollectionElement] func(i T, index int) any
```

making the listMapFunction generic
also, we can use `any` instead of `interface{}` everywhere, since both are equivalent and it is easier on the eyes / less verbose

Reply:
@aswath-s-tw Changes done!